pkg/lib/aws/aws.go

```diff
@@ -43,6 +44,14 @@ func NewFromCreds(region string, accessKeyID string, secretAccessKey string) (*C
 	return New(region, creds)
 }
 
+func NewFromClientS3Path(s3Path string, awsClient *Client) (*Client, error) {
+	if !awsClient.IsAnonymous {
+		return NewFromCredsS3Path(s3Path, *awsClient.AccessKeyID(), *awsClient.SecretAccessKey())
```

Comment:
Should we be safe and check that `awsClient.AccessKeyID()` and  `awsClient.SecretAccessKey()` are not nil? And if so, what do you think we should we do in that case?